tests/test_scheduler.py

```diff
@@ -263,19 +263,54 @@ async def coro(fut):
     fut3 = asyncio.Future()
 
     await scheduler.spawn(coro(fut1))
+    assert scheduler.active_count == 1
     assert scheduler.pending_count == 0
 
     await scheduler.spawn(coro(fut2))
+    assert scheduler.active_count == 1
     assert scheduler.pending_count == 1
 
     with pytest.raises(asyncio.TimeoutError):
         # try to wait for 1 sec to add task to pending queue
         async with timeout(1):
             await scheduler.spawn(coro(fut3))
 
+    assert scheduler.active_count == 1
     assert scheduler.pending_count == 1
 
 
+async def test_scheduler_concurrency_pending_limit(make_scheduler):
+    scheduler = await make_scheduler(limit=1, pending_limit=1)
+
+    async def coro(fut):
+        await fut
+
+    futures = [asyncio.Future() for _ in range(3)]
+    jobs = []
+
+    async def spawn():
+        for fut in futures:
+            jobs.append(await scheduler.spawn(coro(fut)))
+
+    asyncio.create_task(spawn())
```

Comment:
This task should probably be assigned to a variable and awaited later on (presumably just before the job.wait() calls).

I'm a little surprised the tests don't error due to an unawaited task..

Reply:
It's true, thank you. Done